packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { IStyle, ITheme } from '../../Styling';
+import { IStyleFunctionOrObject } from '../../Utilities';
+
+export interface IShimmer {}
+
+/**
+ * Shimmer component props.
+ */
+export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
+  /**
+   * Optional callback to access the IShimmer interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IShimmer | null) => void;
+
+  /**
+   * Sets the width of the shimmer wave wrapper in percentages relative to the containig parent element.
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the shimmer wave wrapper to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
```

Comment:
When would this be used? Shouldn't there be some callback to say when the animation is completed?

I assume you'd say true when you want it to fade out?

Reply:
@dzearing When Shimmer is used as a wrapper of the actual content this prop will be used by the host application to trigger the fadeOut of the animation / fadeIn of the content / and at the end I am removing the animation div from the DOM. So basically the divs holding the content are already rendered without the content and hidden behind the shimmer animation and I am using the prop to swap them.